src/GafferOSL/ShadingEngine.cpp

```diff
@@ -642,14 +803,142 @@ class GafferBatchedRendererServices : public OSL::BatchedRendererServices<WidthT
 			return false;
 		}
 
+		void pointcloud_search( BatchedShaderGlobals *sg, ustringhash filename, const void *wideCenterVoid, Wide<const float> wideRadius, int maxPoints, bool sort, typename OSL::BatchedRendererServices<WidthT>::PointCloudSearchResults &results ) override
```

Comment:
Think we'll need to guard against negative `maxPoints` values here and in `RendererServices::pointcloud_search()`, or a dodgy bit of OSLCode with `maxPoints = -1` will result in a SIGABRT.

Reply:
Done in https://github.com/GafferHQ/gaffer/pull/6470/commits/5c117b99e1c4e92efd2e77988d27ba145070c641. I originally added a test for this too, but that exposed a bug in OSL itself for 1.14+, so I haven't included the test in the commit.